### PR TITLE
add support for lhttpc pools

### PR DESCRIPTION
**Pull request by juantascon**

### CODE REVIEW

File: src/erlcloud_httpc.erl [hunk 1]
```diff
@@ -43,8 +43,13 @@ request(URL, Method, Hdrs, Body, Timeout,
     when is_function(F, 6) ->
     F(URL, Method, Hdrs, Body, Timeout, Config).
 
-request_lhttpc(URL, Method, Hdrs, Body, Timeout, _Config) ->
-    lhttpc:request(URL, Method, Hdrs, Body, Timeout, []).
+request_lhttpc(URL, Method, Hdrs, Body, Timeout, #aws_config{lhttpc_pool = Pool}) ->
```
Comment (motobob) [inline src/erlcloud_httpc.erl:48]: a funs with one case/if should be using function clauses.

``` erlang
request_lhttpc(URL, Method, Hdrs, Body, Timeout, #aws_config{lhttpc_pool = undefined}) ->
    lhttpc:request(URL, Method, Hdrs, Body, Timeout, []);
request_lhttpc(URL, Method, Hdrs, Body, Timeout, #aws_config{lhttpc_pool = Pool}) ->
    lhttpc:request(URL, Method, Hdrs, Body, Timeout, [{pool, Pool}]).
```

**Comment by motobob** [comment]
what is the use case to have that? are you planning to have a library to be used with multiple lhttpc pools at the same time?
how will it help given that AWS has it's own API throttling on a per service basis?

**Comment by motobob** [comment]
@juantascon please address the style comment and then it's 👍

**Comment by nalundgaard** [comment]
@motobob shouldn't this be PRed to the mainline erlcloud? Why would we add to the divergence of the AL fork for this?

**Comment by motobob** [comment]
yep. to be added to upstream too.

**Comment by juantascon** [comment]
I'll fix the readme and will submit again but I'm not sure I got that last part, should I pullrq this to github.com/erlcloud/erlcloud ?
@motobob ^^

**Comment by motobob** [comment]
yes. cherry-pick the commit on top or erlcloud/erlcloud/master. do not send a PR from this branch directly to upstream as the diff is crazy big

**Comment by juantascon** [comment]
submitted here: https://github.com/erlcloud/erlcloud/pull/300

**Comment by motobob** [comment]
@juantascon sorry for confusion. i meant PRs to both forks but that's fine. i'll handle the magic.

**Pull request closed without merging**